### PR TITLE
ARTEMIS-2391 static address may be auto-deleted

### PR DESCRIPTION
**Pull request by jbertram**

### CODE REVIEW

**Comment by michaelandrepearce** [comment]
Will look at this next week. Just quickly in laymans terms whats the noticed issue trying to stop or prevent. Realise it might be evident when i review next week, as im sure tests added probably define it.

**Comment by jbertram** [comment]
@michaelandrepearce, if you statically define an address (e.g. in broker.xml or via core API) and that address has no bindings it may be inadvertently removed by `org.apache.activemq.artemis.core.postoffice.impl.PostOfficeImpl.AddressQueueReaper`.

**Comment by michaelandrepearce** [comment]
@jbertram thanks, make sense. +1 (though you've already merged it)

**Comment by toolkit** [comment]
@michaelandrepearce - IG saw this issue in 7.3. 
We were advised to set address-queue-scan-period to -1 as a temporary workaround.

**Comment by michaelandrepearce** [comment]
@toolkit i heard from @henock friday evening

**Comment by toolkit** [comment]
A question on this. If addressQueueReaperPeriod is set to -1, how will an existing addressQueueReaperRunnable be stopped?

**Comment by toolkit** [comment]
@jbertram - one question on this. It looks like  addressQueueReaperRunnable.stop() is only called if this object is not null, but also only if addressQueueReaperPeriod is > 0.

Assuming that the broker XML configuration file is only modified, and the broker is not restarted, then does this mean that the  addressQueueReaperRunnable cannot be disabled by setting it to -1 ?

**Comment by jbertram** [comment]
@toolkit, that's correct. The broker will need to be restarted in order to stop the ` addressQueueReaperRunnable` and prevent it from starting again. You'll note in [the documentation](http://activemq.apache.org/components/artemis/documentation/latest/config-reload.html) that only a handful of settings actually get reloaded when changed via `broker.xml` at runtime.
